torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -103,29 +104,30 @@ std::unordered_set<c10::DeviceIndex> getLocalDevices(
 // 1) checks there is no duplication in the devices field
 // 2) checks all local devices in the deviceSet are included in deviceOpt
 void checkValidDevicesOption(
-    const std::unordered_set<c10::DeviceIndex>& deviceSet,
-    const std::vector<c10::DeviceIndex>& deviceOpt) {
-  std::unordered_set<c10::DeviceIndex> optsDeviceSet(
-      deviceOpt.begin(), deviceOpt.end());
+    const std::unordered_set<c10::Device>& deviceSet,
+    const std::vector<c10::Device>& deviceOpt) {
+  std::unordered_set<c10::Device> optsDeviceSet;
+  for (const c10::Device& dev : deviceOpt) {
+    optsDeviceSet.insert(dev);
+  }
 
   // no duplications are allowed in opts_.devices
   TORCH_CHECK(
       deviceOpt.size() == optsDeviceSet.size(),
       "Detected duplication in TensorPipeRpcBackendOptions devices field.");
 
   // opts_.devices must be a superset of local devices in reverseDeviceMaps_
-  std::vector<c10::DeviceIndex> cut;
-  std::set_difference(
```

Comment:
I believe `set_difference` requires its inputs to be sorted, so I don't understand how this used to work with `unordered_set` which, as its name says, isn't sorted...